common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -192,7 +194,19 @@ public TransportClient createClient(String remoteHost, int remotePort)
           logger.info("Found inactive connection to {}, creating a new one.", resolvedAddress);
         }
       }
-      clientPool.clients[clientIndex] = createClient(resolvedAddress);
+      if (System.currentTimeMillis() - clientPool.lastConnectionFailed[clientIndex]
```

Comment:
clientIndex here is just a random number, so this isn't always going to work if the last task that failed got a different clientIndex, right?

Reply:
I think I get your point.
Do you means that, we need define a new exception type, when retryBlockFetcher catch this exception, the retry count should not increase?

Reply:
no, I mean you are using a random number here and the last task that failed might not have matched.  Don't you need an indicator over the entire clientPool?